### PR TITLE
feat: simulate raid

### PR DESCRIPTION
**Pull request by edisontim**

### CODE REVIEW

File: client/src/ui/modules/military/battle-view/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { ClientComponents } from "@/dojo/createClientComponents";
+import { ArmyInfo } from "@/hooks/helpers/useArmies";
+import { Battle, EternumGlobalConfig, Health, Troops as SdkTroops, TroopConfig } from "@bibliothecadao/eternum";
```
Comment (ponderingdemocritus) [inline client/src/ui/modules/military/battle-view/utils.tsx:3]: EternumGlobalConfig - use config manager

File: client/src/ui/modules/military/battle-view/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  return (health.current * BigInt(10_000)) / health.lifetime;
+}
+
+export const getMaxResourceAmountStolen = (
+  attackerArmy: ArmyInfo | undefined,
+  defenderArmy: ArmyInfo | undefined,
+  troopConfig: ComponentValue<ClientComponents["TroopConfig"]["schema"]>,
+) => {
+  if (!attackerArmy) return 0;
+  const attackingTroops = getTotalTroops(attackerArmy.troops) / EternumGlobalConfig.resources.resourcePrecision;
```
Comment (ponderingdemocritus) [inline client/src/ui/modules/military/battle-view/utils.tsx:53]: precision move to manager

File: client/src/ui/modules/military/battle-view/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  const totalCrossbowmanStrength = BigInt(troopConfig.crossbowman_strength) * crossbowmanCount;
+
+  return totalKnightStrength + totalPaladinStrength + totalCrossbowmanStrength;
+}
+
+function percentageLeft(health: ComponentValue<ClientComponents["Health"]["schema"]>): bigint {
+  if (health.lifetime === 0n) {
+    return 0n;
+  }
+  return (health.current * BigInt(10_000)) / health.lifetime;
```
Comment (ponderingdemocritus) [inline client/src/ui/modules/military/battle-view/utils.tsx:44]: is this precision?

**Review by ponderingdemocritus — COMMENTED** [review]
amazing work on this - huge upgrade. Just some hard coded config values need fixing.

File: client/src/ui/modules/military/battle-view/utils.tsx (hunk 3 above)
Comment (edisontim) [reply client/src/ui/modules/military/battle-view/utils.tsx:44]: My bad, this was a percentage value. Changing right now

**Comment by edisontim** [comment]
> amazing work on this - huge upgrade. Just some hard coded config values need fixing.

@credence0x should take the credit for this, I've merely re-used his simulateBattle script haha

File: client/src/dojo/modelManager/ConfigManager.ts [hunk 4]
```diff
@@ -509,6 +509,10 @@ export class ClientConfigManager {
     return EternumGlobalConfig.resources.resourcePrecision;
   }
 
+  getResourceMultiplier() {
+    return EternumGlobalConfig.resources.resourceMultiplier;
```
Comment (ponderingdemocritus) [inline client/src/dojo/modelManager/ConfigManager.ts:513]: this needs to be fetched from RECS not from the config

Comment (aymericdelab) [reply client/src/dojo/modelManager/ConfigManager.ts:513]: resource multiplier isn't present in the state, it's a value we use to set the config

**Comment by aymericdelab** [comment]
lgtm

**Pull request merged**